tests/functional/init/test_init.py

```diff
@@ -708,3 +730,130 @@ def test_init_inside_project_and_skip_profile_setup(
         # skip interactive profile setup
         run_dbt(["init", "--skip-profile-setup"])
         assert len(manager.mock_calls) == 0
+
+
+class TestInitOutsideOfProjectWithSpecifiedProfile(TestInitOutsideOfProjectBase):
+    @mock.patch("dbt.task.init._get_adapter_plugin_names")
+    @mock.patch("click.prompt")
+    def test_init_task_outside_of_project_with_specified_profile(
+        self, mock_prompt, mock_get_adapter, project, project_name, unique_schema, dbt_profile_data
+    ):
+        manager = Mock()
+        manager.attach_mock(mock_prompt, "prompt")
+        manager.prompt.side_effect = [
+            project_name,
+        ]
+        mock_get_adapter.return_value = [project.adapter.type()]
+        run_dbt(["init", "--profile", "test"])
+
+        manager.assert_has_calls(
+            [
+                call.prompt("Enter a name for your project (letters, digits, underscore)"),
+            ]
+        )
+
+        # profiles.yml is NOT overwritten, so assert that the text matches that of the
+        # original fixture
+        with open(os.path.join(project.profiles_dir, "profiles.yml"), "r") as f:
+            assert f.read() == yaml.safe_dump(dbt_profile_data)
+
+        with open(os.path.join(project.project_root, project_name, "dbt_project.yml"), "r") as f:
+            assert (
+                f.read()
+                == f"""
```

Comment:
Can we make this a fixture in a constant at file top or in a `fixture.py`?

Reply:
Checking the other init tests shows that all of them use this pattern. Its probably worth getting more opinions before changing it.